llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2165,15 +2171,9 @@ MemoryDepChecker::isDependent(const MemAccessInfo &A, unsigned AIdx,
 
   // Negative distances are not plausible dependencies.
   if (SE.isKnownNonPositive(Dist)) {
-    if (SE.isKnownNonNegative(Dist)) {
-      if (HasSameSize) {
-        // Write to the same location with the same size.
-        return Dependence::Forward;
-      }
-      LLVM_DEBUG(dbgs() << "LAA: possibly zero dependence difference but "
-                           "different type sizes\n");
-      return Dependence::Unknown;
-    }
+    if (SE.isKnownNonNegative(Dist))
+      // Write to the same location with the same size.
+      return Dependence::Forward;
```

Comment:
```suggestion
    if (SE.isKnownNonNegative(Dist)) {
      // Write to the same location with the same size.
      return Dependence::Forward;
    }
```

nit: {} are recommended for all blocks more than one line IIRC